### DISCUSSION TITLE
Custom webpack configuration in `next.config.js` not invoked

### DISCUSSION THREAD

**Discussion by filipposarzana** [vercel/next.js#21440] [Help]
**What version of Next.js are you using?**

10.0.5

**What version of Node.js are you using?**

12.19.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

next build

**Describe the Bug**

Adding a custom `webpack: (config) =>` to `next.config.js` is not working as it's not invoked even though it's reported here https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config

I also tried with `@next/bundle-analyzer` and it's not working either as it's setting `webpack` which is again not invoked.

**Expected Behavior**

To be able to specify custom webpack configuration

**To Reproduce**

Just folowing this https://github.com/vercel/next.js/tree/canary/packages/next-bundle-analyzer, it won't work

**Comment by monokrome** [comment]
Same issue here. We can't run our app in production because the config's **webpack** export never gets called. If you drop a console.log at the top of the function, it never happens.

Furthermore, **next build** doesn't respect *tsconfig.json* aliases like `npm run dev` does. How can such a big core feature of next be completely broken for this long?

Apparently this doesn't even break builds:

```javascript
module.exports = {
  webpack: function (config) { process.exit(0); },
};

```

**Comment by justintemps** [comment]
@filipposarzana did you ever figure this out?
